File: src/main/java/bad/robot/radiate/ui/DesktopMode.java
package bad.robot.radiate.ui;

import javax.swing.*;
import java.awt.*;

import static java.awt.Color.darkGray;

@Deprecated
/* aka resizable screen mode */
class DesktopMode implements ScreenMode {
    private final Rectangle bounds;

    public DesktopMode(Rectangle bounds) {
        this.bounds = bounds;
    }

    @Override
    public void accept(JFrame frame) {
        frame.setSize((int) (bounds.width * 0.80), (int) (bounds.height * 0.80));
        frame.setLocation(bounds.x, bounds.y);
        frame.getContentPane().setBackground(darkGray);
    }
}

File: src/main/scala/bad/robot/radiate/ui/DesktopMode.scala
package bad.robot.radiate.ui

import java.awt.Color.darkGray
import java.awt._
import javax.swing._

/** aka resizable screen mode */
class DesktopMode(bounds: Rectangle) extends ScreenMode {

  def accept(frame: JFrame) {
    frame.setSize((bounds.width * 0.80).toInt, (bounds.height * 0.80).toInt)
    frame.setLocation(bounds.x, bounds.y)
    frame.getContentPane.setBackground(darkGray)
  }
}